daemon/src/auto_rollover.rs

```diff
@@ -0,0 +1,194 @@
+use crate::address_map::AddressMap;
+use crate::address_map::Stopping;
+use crate::cfd_actors::append_cfd_state;
+use crate::connection;
+use crate::db;
+use crate::db::load_cfd;
+use crate::model::cfd::CannotRollover;
+use crate::model::cfd::CfdState;
+use crate::model::cfd::CfdStateCommon;
+use crate::model::cfd::OrderId;
+use crate::monitor;
+use crate::monitor::MonitorParams;
+use crate::oracle;
+use crate::projection;
+use crate::rollover_taker;
+use crate::Tasks;
+use anyhow::Result;
+use async_trait::async_trait;
+use maia::secp256k1_zkp::schnorrsig;
+use std::time::Duration;
+use xtra::Actor as _;
+use xtra::Address;
+use xtra_productivity::xtra_productivity;
+
+pub struct Actor<O, M> {
+    db: sqlx::SqlitePool,
+    oracle_pk: schnorrsig::PublicKey,
+    projection_actor: Address<projection::Actor>,
+    conn_actor: Address<connection::Actor>,
+    monitor_actor: Address<M>,
+    oracle_actor: Address<O>,
+    n_payouts: usize,
+
+    rollover_actors: AddressMap<OrderId, rollover_taker::Actor>,
+
+    tasks: Tasks,
+}
+
+impl<O, M> Actor<O, M> {
+    pub fn new(
+        db: sqlx::SqlitePool,
+        oracle_pk: schnorrsig::PublicKey,
+        projection_actor: Address<projection::Actor>,
+        conn_actor: Address<connection::Actor>,
+        monitor_actor: Address<M>,
+        oracle_actor: Address<O>,
+        n_payouts: usize,
+    ) -> Self {
+        Self {
+            db,
+            oracle_pk,
+            projection_actor,
+            conn_actor,
+            monitor_actor,
+            oracle_actor,
+            n_payouts,
+            rollover_actors: AddressMap::default(),
+            tasks: Tasks::default(),
+        }
+    }
+}
+
+#[xtra_productivity]
+impl<O, M> Actor<O, M>
+where
+    M: xtra::Handler<monitor::StartMonitoring>,
+    O: xtra::Handler<oracle::MonitorAttestation> + xtra::Handler<oracle::GetAnnouncement>,
+{
+    async fn handle(&mut self, _msg: AutoRollover, ctx: &mut xtra::Context<Self>) -> Result<()> {
```

Comment:
nit: this would be much more readable if the message was a verb, e.g. `StartAutoRollover`.

Reply:
Agreed, feel free to send a follow-up :D